File: Roboti/app/src/main/java/cz/muni/fi/umimecesky/roboti/adapterlistener/LevelAdapter.java
package cz.muni.fi.umimecesky.roboti.adapterlistener;

import android.app.Activity;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.akexorcist.roundcornerprogressbar.RoundCornerProgressBar;

import java.util.List;

import cz.muni.fi.umimecesky.roboti.R;
import cz.muni.fi.umimecesky.roboti.pojo.RaceConcept;

public class LevelAdapter extends BaseAdapter {

    private List<RaceConcept> list;
    private Activity activity;

    public LevelAdapter(Activity activity, List<RaceConcept> list) {
        super();
        this.activity = activity;
        this.list = list;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        final ViewHolder holder;

        if (convertView == null) {
            convertView = LayoutInflater.from(activity).inflate(R.layout.column_category_race, parent, false);

            holder = new ViewHolder();
            holder.sectionName = (TextView) convertView.findViewById(R.id.sectionName);
            holder.progressBar = (RoundCornerProgressBar) convertView.findViewById(R.id.levelProgress);
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }

        RaceConcept raceConcept = list.get(position);
        Log.d("race concept", raceConcept + "");
        holder.sectionName.setText(raceConcept.getName());
        progressBarSetup(holder.progressBar,raceConcept);
        return convertView;
    }

    private void progressBarSetup(RoundCornerProgressBar progressBar, RaceConcept raceConcept) {
        int currentLevel = raceConcept.getCurrentLevel();
        progressBar.setProgress(currentLevel);
        progressBar.setSecondaryProgress(currentLevel + 1);
    }

    private String buildLevelsString(int maxLevel) {
        StringBuilder builder = new StringBuilder();
        for (int i = 1; i <= maxLevel; i++) {
            builder.append(i).append(" ");
        }
        return builder.toString();
    }

    private class ViewHolder {
        TextView sectionName;
        RoundCornerProgressBar progressBar;
    }

    @Override
    public int getCount() {
        return list.size();
    }

    @Override
    public Object getItem(int position) {
        return list.get(position);
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }
}

File: app/src/main/java/cz/muni/fi/umimecesky/game/robots/LevelAdapter.kt
/*
 * Copyright (c) 2018 Marek Sabo
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *                http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package cz.muni.fi.umimecesky.game.robots

import androidx.recyclerview.widget.RecyclerView
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.akexorcist.roundcornerprogressbar.RoundCornerProgressBar
import cz.muni.fi.umimecesky.R
import cz.muni.fi.umimecesky.game.shared.model.RaceConcept
import cz.muni.fi.umimecesky.game.shared.util.GuiUtil.inflate

class LevelAdapter(private val items: List<RaceConcept>,
                   private val listener: (RaceConcept) -> Unit) :
        RecyclerView.Adapter<LevelAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
            ViewHolder(parent.inflate(R.layout.column_category_race))

    override fun onBindViewHolder(holder: ViewHolder, position: Int) =
            holder.bind(items[position], listener)

    override fun getItemCount(): Int = items.size

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        private val sectionName = itemView.findViewById<TextView>(R.id.sectionName)
        private val currentLevel = itemView.findViewById<TextView>(R.id.currentLevel)
        private val progressBar = itemView.findViewById<RoundCornerProgressBar>(R.id.levelProgress)

        fun bind(item: RaceConcept, listener: (RaceConcept) -> Unit) = with(itemView) {
            sectionName.text = item.name
            val currentLevelNumber = item.currentLevel
            currentLevel.text = currentLevelNumber.toString()
            progressBar.max = (item.numberOfLevels - 1).toFloat()
            progressBar.progress = (currentLevelNumber - 1).toFloat()
            progressBar.secondaryProgress = currentLevelNumber.toFloat()
            setOnClickListener { listener(item) }
        }
    }

}

